src/Storages/StorageDistributed.cpp

```diff
@@ -893,10 +893,26 @@ void StorageDistributed::rename(const String & new_path_to_table_data, const Sto
 }
 
 
-size_t StorageDistributed::getRandomShardIndex(size_t shard_num)
+size_t StorageDistributed::getRandomShardIndex(const Cluster::ShardsInfo & shards)
 {
     std::lock_guard lock(rng_mutex);
```

Comment:
Ideally we can lock less)

Reply:
You're right - we need it only for `rng`